metadata_mapper/mappers/oai/samvera_mapper.py

```diff
@@ -3,12 +3,70 @@
 
 class SamveraRecord(OaiRecord):
 
+    TYPES = {
+        "img": "image",
+        "txt": "text"
+    }
+
+    TYPES_BASEURL = "http://id.loc.gov/vocabulary/resourceTypes/"
+
     def UCLDC_map(self):
         return {
-            'is_shown_at': self.source_metadata.get('isShownAt'),
-            'is_shown_by': self.source_metadata.get('isShownBy')
+            "alternativeTitle": self.map_alternative_title,
+            "type": self.map_type,
         }
 
+    def map_alternative_title(self) -> list:
+        value = self.source_metadata.get("alternative")
+
+        if isinstance(value, list):
+            return value
+
+        return [value]
+
+    def map_is_shown_at(self) -> str:
+        value = self.source_metadata.get("isShownAt")
+
+        if isinstance(value, list):
+            return value[0]
+        elif isinstance(value, str):
+            return value
+        return None
+
+    def map_is_shown_by(self) -> str:
+        value = self.source_metadata.get("object")
+
+        if isinstance(value, list):
+            return value[0]
+        elif isinstance(value, str):
+            return value
+        return None
+
+    def map_type(self) -> list:
```

Comment:
I think this function should also be moved to the OAI mapper. Or was there a reason to put it here? The logic is hairy in the legacy harvester, but it looks to me like it could be in the base class.